lib/opal/nodes/args/initialize_kwargs.rb

```diff
@@ -0,0 +1,32 @@
+require 'opal/nodes/base'
+
+module Opal
+  module Nodes
+    # An abstract node responsible for validating
+    # keyword arguments in the post-splat list of arguments.
+    #
+    # @see PostArgsNode
+    #
+    class InitializeKwargsNode < Base
+      def compile
+        unless scope.kwargs_initialized
+          initialize_kw_args
+        end
+      end
+
+      def initialize_kw_args
+        helper :hash2
+
+        line "if ($kwargs && !$kwargs.$$is_hash) {"
+        line "  throw Opal.ArgumentError.$new('expected kwargs');"
+        line "}"
+
+        line "if (!$kwargs || !$kwargs.$$is_hash) {"
+        line "  $kwargs = $hash2([], {});"
+        line "}"
```

Comment:
I think this check should use `== null` that includes `null` and `undefined`.

```js
if ($kwargs != null && !$kwargs.$$is_hash) {
  throw Opal.ArgumentError.$new('expected kwargs');
}
if ($kwargs == null || !$kwargs.$$is_hash) {
  $kwargs = $hash2([], {});
}
```

or 


```js
if ($kwargs == null || !$kwargs.$$is_hash) {
  if ($kwargs == null) {
    $kwargs = $hash2([], {});
  } else {
    throw Opal.ArgumentError.$new('expected kwargs');
  }
}
```

not sure which is better, the last one saves a check for the happy path of actually passing kwargs.